packages/opentelemetry-plugin-http/src/utils.ts

```diff
@@ -84,7 +84,8 @@ export class Utils {
   static hasExpectHeader(options: RequestOptions | url.URL): boolean {
     return !!(
       (options as RequestOptions).headers &&
-      (options as RequestOptions).headers!.Expect
+      ((options as RequestOptions).headers!.Expect ||
```

Comment:
This still doesn't cover all allowed cases as it works also to use e.g. `ExPeCt`. Clearly the most common cases are handled.